baselines/fedht/fedht/main.py

```diff
@@ -0,0 +1,164 @@
+"""Run main for fedht baseline."""
+
+import pickle
+import random
+import torch
+
+import flwr as fl
+import hydra
+import numpy as np
+from flwr.common import NDArrays, ndarrays_to_parameters
+from flwr.server.strategy.strategy import Strategy
+from flwr_datasets import FederatedDataset
+from flwr_datasets.partitioner import PathologicalPartitioner
+from omegaconf import DictConfig
+from torch.utils.data import DataLoader
+
+from fedht.client import generate_client_fn_mnist, generate_client_fn_simII
+from fedht.fedht import FedHT
+from fedht.model import LogisticRegression
+from fedht.server import fit_round, get_evaluate_fn
+from fedht.utils import MyDataset, sim_data
+
+
+@hydra.main(config_path="conf", config_name="base_mnist", version_base=None)
+def main(cfg: DictConfig):
+    """Run main file for fedht baseline.
+
+    Parameters
+    ----------
+    cfg : DictConfig
+        Config file for federated baseline; read from fedht/conf.
+    """
+    # set seed
+    random.seed(2024)
+
+    # this vs. setting in cfg; what is preferred?
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+    if cfg.data == "mnist":
+        # set up mnist data
+        # set partitioner
+        partitioner = PathologicalPartitioner(
+            num_partitions=cfg.num_clients,
+            partition_by="label",
+            num_classes_per_partition=2,
+            class_assignment_mode="first-deterministic",
+        )
+
+        # load MNIST data
+        num_features = 28 * 28
+        num_classes = 10
+        dataset = FederatedDataset(dataset="mnist", partitioners={"train": partitioner})
+        test_dataset = dataset.load_split("test").with_format("numpy")
+        testloader = DataLoader(test_dataset, batch_size=cfg.batch_size, shuffle=False)
+
+        # define model
+        model = LogisticRegression(num_features, num_classes)
```

Comment:
Let's move this inside `generate_client_fn_mnist`?

Reply:
moved inside generate_client_fn_mnist